agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/MainEntryPoint.java

```diff
@@ -134,19 +135,34 @@ private static void logErrorMessage(Logger startupLogger, String message, boolea
             try {
                 // IF the startupLogger failed to be initialized due to configuration syntax error, try initializing it here
                 Path agentPath = new File(bootstrapURL.toURI()).toPath();
-                startupLogger = configureLogging(new SelfDiagnostics(), agentPath);
+                SelfDiagnostics selfDiagnostics = new SelfDiagnostics();
+                selfDiagnostics.file.path = ConfigurationBuilder.overlayWithEnvVar(
+                        ConfigurationBuilder.APPLICATIONINSIGHTS_SELF_DIAGNOSTICS_FILE_PATH, selfDiagnostics.file.path);
+                startupLogger = configureLogging(selfDiagnostics, agentPath);
                 if (isFriendlyException) {
                     startupLogger.error(message);
                 } else {
                     startupLogger.error(message, t);
                 }
-            } catch (Throwable e) {
-                // If the startupLogger still have some issues being initialized, just print the error stack trace
+            } catch (Throwable ignored) {
+                // this is a last resort in cases where the JVM doesn't have write permission to the directory where the agent lives
+                // and the user has not specified APPLICATIONINSIGHTS_SELF_DIAGNOSTICS_FILE_PATH
```

Comment:
file.path is currently default to "applicationinsights.log". for azure function, it doesn't have the write permission in the agent jar folder. can we make azure function or other attaches to default to "/var/log/applicationinsights.log"?

Reply:
of course, we can always ask them to provide a value for this env var or declare it in the json config. 
this env var will have a higher precedence over the json config, right? similar to other env vars' precedence.

Reply:
this shouldn't happen for users, since they can't control the json file, and so can't mess it up accidentally and cause it to fail to start

also will still log to `/tmp/applicationinsights.log` now which is better than before (just not the more ideal `/var/log/applicationinsights.log`, trying to balance the amount of special cases we add to the code for different compute environments)